examples/all-clusters-app/all-clusters-common/src/WhmManufacturer.cpp

```diff
@@ -0,0 +1,270 @@
+                                            Optional<int16_t> temporarySetpoint, Optional<chip::Percent> targetPercentage,
+                                            Optional<chip::Percent> targetReheat)
+{
+    return Status::Success;
+}
+
+Status WhmManufacturer::BoostCommandCancelled()
+{
+    return Status::Success;
+}
+
+void WhmManufacturer::BoostCommandFinished() {}
+
+WaterHeaterManagementDelegate * GetWhmDelegate()
+{
+    WhmManufacturer * mn = GetWhmManufacturer();
+    VerifyOrDieWithMsg(mn != nullptr, AppServer, "WhmManufacturer is null");
+
+    WaterHeaterManagementDelegate * wg = mn->GetWhmDelegate();
+    VerifyOrDieWithMsg(wg != nullptr, AppServer, "WhmDelegate is null");
+
+    return wg;
+}
+
+void SetTestEventTrigger_BasicInstallationTestEvent()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    // Simulate installation in a 100L tank full of water at 20C, with a target temperature of 60C, in OFF mode
+    dg->SetTankVolume(100);
+    dg->SetTargetWaterTemperature(6000);
+    dg->SetHeaterTypes(BitMask<WaterHeaterTypeBitmap>(WaterHeaterTypeBitmap::kImmersionElement1));
+    dg->DrawOffHotWater(100, 2000);
+}
+
+void SetTestEventTrigger_BasicInstallationTestEventClear() {}
+
+void SetTestEventTrigger_WaterTemperature20CTestEvent()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    // Simulate 100% of the water in the tank being at 20C
+    dg->SetWaterTemperature(2000);
+}
+
+void SetTestEventTrigger_WaterTemperature61CTestEvent()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    // Simulate 100% of the water in the tank being at 61C
+    dg->SetWaterTemperature(6100);
+}
+
+void SetTestEventTrigger_WaterTemperature66CTestEvent()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    // Simulate 100% of the water in the tank being at 66C
+    dg->SetWaterTemperature(6600);
+}
+
+void SetTestEventTrigger_ManualModeTestEvent()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    // Simulate the Water Heater Mode being set to MANUAL
+    dg->SetWaterHeaterMode(WaterHeaterManagementDelegate::ModeManual);
+}
+
+void SetTestEventTrigger_OffModeTestEvent()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    // Simulate the Water Heater Mode being set to OFF
+    dg->SetWaterHeaterMode(WaterHeaterManagementDelegate::ModeOff);
+}
+
+void SetTestEventTrigger_DrawOffHotWaterTestEvent()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    // Simulate drawing off 25% of the tank volume of hot water, replaced with water at 20C
+    dg->DrawOffHotWater(25, 2000);
+}
+
+} // namespace WaterHeaterManagement
+} // namespace Clusters
+} // namespace app
+} // namespace chip
+
+using namespace chip::app::Clusters::WaterHeaterManagement;
+
+bool HandleWaterHeaterManagementTestEventTrigger(uint64_t eventTrigger)
```

Comment:
Should these test event triggers be put into a separate file to make it easier to split out when this example is added to energy-management app in the same way you did the others?

Reply:
This can be done later.